### PR TITLE
Development admin accessibility backend

### PR DESCRIPTION
**Pull request by Mahmoud-Emad**
### Description

Administrators can now apply for vacations on behalf of other users in the same office using the new endpoint. The endpoint has been created based on the provided code specifications.

- Endpoint Name: AdminApplyVacationForUserApiView
- HTTP Method: POST
- Route: /api/vacations/admin-post-for-user/{user_id}/
- Permission Required: Admin

### payload

```json
{
  "reason": "annual_leaves",
  "from_date": "2024-03-05",
  "end_date": "2024-03-05"
}
```

## Apply for Vacation on Behalf of Another User
This function allows the administrator to apply for a vacation on behalf of another user, but only if they work in the same office.

### Parameters:
- **user_id** (`int`): The ID of the user for whom the vacation is being requested.
- **reason** (`str`): A string describing the reason for the vacation request.
- **from_date** (`datetime`): The starting date of the vacation.
- **end_date** (`datetime`): The ending date of the vacation.

image

### CODE REVIEW

**Comment by xmonader** [comment]
why admin-post-for-user?

post /vacations/{user_id} and you know the sender, if the sender is admin, allow it

**Comment by Mahmoud-Emad** [comment]
> why admin-post-for-user?
> 
> post /vacations/{user_id} and you know the sender, if the sender is admin, allow it

I have another view under `vacations/ID` can it be `vacations/admin/ID`?

File: server/cshr/views/vacations.py [hunk 1]
```diff
@@ -755,3 +755,132 @@ def get(self, request: Request) -> Response:
...
+            reason = serializer.validated_data.get('reason')
+            if reason not in known_reasons:
+                formatted_string = ', '.join(known_reasons[:-1]) + f' and {known_reasons[-1]}'
+                return CustomResponse.bad_request(message=f"reason {reason} is not valid, the available resons are {formatted_string}")
+
+            from_date = serializer.validated_data.get('from_date')
+            end_date = serializer.validated_data.get('end_date')
+
+            # Check if there are pinding vacations in the same day
+            pinding_requests = Vacation.objects.filter(
```
Comment (maayarosama) [inline server/cshr/views/vacations.py]: pending_requests

File: server/cshr/views/vacations.py [hunk 2]
```diff
@@ -755,3 +755,132 @@ def get(self, request: Request) -> Response:
...
+                            reason
+                            for reason in REASON_CHOICES
+                            if reason != REASON_CHOICES.PUBLIC_HOLIDAYS
+                        ],
+                    },
+                )
+
+            curr_balance = getattr(user_reason_balance, reason)
+
+            pinding_vacations = Vacation.objects.filter(
```
Comment (maayarosama) [inline server/cshr/views/vacations.py]: pending_vacations

**Review by maayarosama — APPROVED** [review]

**Pull request merged**